node/libs/roles/src/validator/messages/consensus.rs

```diff
@@ -74,25 +76,48 @@ impl Default for Fork {
 /// A struct that represents a set of validators. It is used to store the current validator set.
 /// We represent each validator by its validator public key.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct ValidatorSet {
+pub struct ValidatorCommittee {
     vec: Vec<validator::PublicKey>,
-    map: BTreeMap<validator::PublicKey, usize>,
+    indexes: BTreeMap<validator::PublicKey, usize>,
+    weights: Vec<usize>,
 }
 
-impl ValidatorSet {
-    /// Creates a new ValidatorSet from a list of validator public keys.
-    pub fn new(validators: impl IntoIterator<Item = validator::PublicKey>) -> anyhow::Result<Self> {
+impl ValidatorCommittee {
+    /// Maximum validator weight
+    /// 100.00%
```

Comment:
why are we hardcoding the precision here? Why not allow arbitrary weights?

Reply:
+1. There's no reason to have a max weight. Just calculate the threshold from the sum of the weights.

Reply:
👍